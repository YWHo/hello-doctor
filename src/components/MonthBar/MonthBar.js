import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import dayjs from 'dayjs';
import PropTypes from 'prop-types';
import calendarIcon from '../../assets/calendar_icon.svg';
import MonthCalendar from '../MonthCalendar';
import { getSelectedDate, getShowingCalendar } from '../../state/selectors';
import { toggleShowingCalendar } from '../../state/actions';

const Container = styled.div`
  background-color: #177d91;
  padding: 0.4rem;
`;

const ContainerTop = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 39px;
`;

const Month = styled.div`
  color: #fff;
  font-family: 'Roboto', sans-serif;
  font-size: 16px;
`;

const Icon = styled.img`
  margin-bottom: 0.1rem;
`;

export function MonthBar(props) {
  const { selectedDate = dayjs() } = props;
  const month = dayjs(selectedDate)
    .format('MMMM')
    .toUpperCase();

  return (
    <Container>
      <ContainerTop onClick={() => onMonthBarClicked(props)}>
        <Month>{month}</Month>
        <Icon src={calendarIcon} alt='Calendar' />
      </ContainerTop>
      <MonthCalendar />
    </Container>
  );
}

function onMonthBarClicked(props) {
  const { dToggleShowingCalendar, isShowingCalendar } = props;
  dToggleShowingCalendar(!isShowingCalendar);
}

MonthBar.propTypes = {
  selectedDate: PropTypes.object
};

const mapStateToProps = state => {
  return {
    isShowingCalendar: getShowingCalendar(state),
    selectedDate: getSelectedDate(state)
  };
};

function mapDispatchToProps(dispatch) {
  return {
    dToggleShowingCalendar: status => dispatch(toggleShowingCalendar(status))
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MonthBar);
