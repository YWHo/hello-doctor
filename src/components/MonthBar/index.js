import MonthBar from './MonthBar';
export default MonthBar;
