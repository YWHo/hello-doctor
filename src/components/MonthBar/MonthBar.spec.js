import React from 'react';
import moment from 'moment';
import { MonthBar } from './MonthBar';
import renderer from 'react-test-renderer';

describe('NavBar', () => {
  it('render correctly', () => {
    const component = renderer.create(
      <MonthBar selectedDate={moment('20190619', 'YYYYMMDD')} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
