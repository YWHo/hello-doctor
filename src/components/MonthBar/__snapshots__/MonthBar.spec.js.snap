// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthBar render correctly 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <styled.div>
      JUNE
    </styled.div>
    <styled.img
      alt="Calendar"
      src="calendar_icon.svg"
    />
  </styled.div>
</styled.div>
`;

exports[`MonthBar render when showing calendar 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div
        $expanded={true}
        key="selMonth_0"
        onClick={[Function]}
        selected={false}
      >
        JUNE
      </styled.div>
      <styled.div
        $expanded={true}
        key="selMonth_1"
        onClick={[Function]}
        selected={true}
      >
        JULY
      </styled.div>
      <styled.div
        $expanded={true}
        key="selMonth_2"
        onClick={[Function]}
        selected={false}
      >
        AUGUST
      </styled.div>
    </styled.div>
    <styled.img
      alt="Calendar"
      src="calendar_icon.svg"
    />
  </styled.div>
</styled.div>
`;
